projects/angular/src/forms/datepicker/date-input.ts

```diff
@@ -181,11 +157,29 @@ export class ClrDateInput extends WrappedFormControl<ClrDateContainer> implement
     }
   }
 
-  private usingClarityDatepicker() {
+  protected datepickerHasFormControl() {
+    return !!this.control;
+  }
+
+  protected setDate(date: Date | string) {
+    if (typeof date === 'string') {
+      date = new Date(date);
+    }
+
+    if (this.previousDateChange !== date) {
+      this.updateDate(date);
+    }
+
+    if (!this.initialClrDateInputValue) {
+      this.initialClrDateInputValue = date;
+    }
+  }
+
+  protected usingClarityDatepicker() {
```

Comment:
```suggestion
  private usingClarityDatepicker() {
```

This is only used in this class.